collections/max_of.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Applies the given selector to all elements in the given collection and returns the max of it
+ *
+ * Example:
+ *
+ * ```ts
+ * import { sumOf } from "./sum_of.ts"
+ * import { assertEquals } from "../testing/asserts.ts"
+ *
+ * const inventory = [
+ *      { name: "mustard", count: 2 },
+ *      { name: "soy", count: 4 },
+ *      { name: "tomato", count: 32 },
+ *  ];
+ * const maxItem = maxOf(inventory, (i) => i.count);
+ * 
+ * assertEquals(maxItem, 32);
+ * ```
+ */
+
+export function maxOf<T>(
+    array: readonly T[],
+    selector: (el: T) => number,
+): number {
+    let maxOf = -Infinity;
+    
+    for (const i of array) {
+        if (selector(i) > maxOf) {
+            maxOf = selector(i);
+        } 
+        else if (Number.isNaN(selector(i))) {
+            return maxOf = NaN
+        }
```

Comment:
`else` on a new line is not really common style (did you do `deno fmt` and `deno lint`?).

Extremely minor nitpick: I would argue that this is not really an `else` case "semantically`. I would personally `continue` in the first `if` and convert the second case to its own `if`.

Also why do you set `maxOf` in the second case? If you are `return`ing anyways, you can just `return Number.NaN` (or even better, your selector result, whatever `NaN` representation that is).